src/flamegraph/mod.rs

```diff
@@ -576,7 +579,40 @@ where
 
     let mut lines: Vec<&str> = input.lines().collect();
     lines.sort_unstable();
-    from_sorted_lines(opt, lines, writer)
```

Comment:
Can this just call `from_reader` at this point? Or can we factor out a function that does the below so we don't end up repeating it? I think that'll also let us bring `fetch_consistent_palette_if_needed` and `save_consistent_palette_if_needed` back inline to said function.

Reply:
Actually, if we change `PaletteMap` to have `String` keys, and thus be stored in `Options`, we may want to fill the `PaletteMap` one level higher in `main` ? This would return `from_reader`/`from_readers` to a simpler form.

I am not sure how this would impact integration tests however. We would probably have to do some additional work in the test itself because `from_reader` is our main interface, right ?

Reply:
Yeah, I think `from_reader` is the primary interface, at least for now. But that takes an `Options`, right?

Reply:
[Done](https://github.com/jonhoo/inferno/pull/77/commits/6a5462050c0d2a8e9d396e35aa1e66a550bc16b2)